modules/fx/arrow-fx/src/main/kotlin/arrow/fx/IORace.kt

```diff
@@ -76,16 +339,10 @@ interface IORace {
     IO.Async(true) { conn, cb ->
       val active = AtomicBooleanW(true)
 
-      val upstreamCancelToken = IO.defer { if (conn.isCanceled()) IO.unit else conn.cancel() }
```

Comment:
this is the actual fix for the issue, removing the upstream cancellation the exception is not thrown anymore